CCAHT-13-create-side-drawer-component

collins-self opened the pull request:
## Added and styled a permanent side drawer
<img width="744" alt="Screenshot 2022-11-18 at 8 55 47 PM" src="https://user-images.githubusercontent.com/95252655/202829120-d0e6cd40-c54a-41bb-b322-6f99051aa538.png">

The side navigation drawer:

Use the [MUI Drawer component](https://mui.com/material-ui/react-drawer/).

Has an item for: dashboard, check in, check out, inventory, and history.

Has an expandable item for settings that when clicked shows options for: general, categories, items, and item attributes.

andlrutt commented on components/Drawer.tsx:
```diff
@@ -0,0 +1,99 @@
+      <Collapse in={open} timeout="auto" unmountOnExit>
+        <List component="div" disablePadding>
+          <ListItemButton sx={{ pl: 9 }}>
+            <ListItemText primary="General" />
```
Can this and the below text be slightly smaller and indented? Just to set it apart visually so it's clear that they are all a part of settings. You can use the `primaryTypographyProps` props to set the same props you'd normally have on a `<Typography>` element.

andlrutt commented on components/Drawer.tsx:
```diff
@@ -0,0 +1,99 @@
+export default function NavigationDrawer() {
+  // Params for the Drawer
+  const drawerWidth = 280
+  // Logo definition //TODO placeholder
```
Can remove TODO?

andlrutt commented on components/Drawer.tsx:
```diff
@@ -0,0 +1,99 @@
+    <Box>
+      <Box sx={{ pr: 3, mt: 5 }}>{logo}</Box>
+      <br />
+      <ListItemButton>
```
A good way to simplify this would be to define another component for one of these button rows, that way we cut down on duplicated code. It'd probably take in an icon, display text, and children (for the case of settings) as props. Let me or Zavier know if you have any questions about this or if you want to pair. We can also get you started in our next standup.

andlrutt commented on components/Drawer.tsx:
```diff
@@ -0,0 +1,99 @@
+          width: drawerWidth,
+        }}
+      >
+        {drawer}
```
I know it's different on the MUI page, but could you replace this with the actual content of `drawer`? It makes it a bit easier to linearly see what's going on.

collins-self replied:
Do you mean instead of defining it then returning it, just define it in line 95?

andlrutt replied:
Yes

andlrutt commented on components/Drawer.tsx:
```diff
@@ -0,0 +1,99 @@
+import AccessTimeIcon from '@mui/icons-material/AccessTime'
+import SettingsIcon from '@mui/icons-material/Settings'
+
+export default function NavigationDrawer() {
```
In the actual file directory, this should be in a `components/NavigationDrawer/index.tsx` file. See [KKB's old repo](https://github.com/hack4impact-utk/keep-knox-beautiful/tree/develop/src/components) for reference.

andlrutt reviewed (changes requested):
Great start! I left a few comments, but it's just about ready :D Good work!

zaviermiller commented on components/Drawer.tsx:
```diff
@@ -0,0 +1,99 @@
+import AccessTimeIcon from '@mui/icons-material/AccessTime'
+import SettingsIcon from '@mui/icons-material/Settings'
+
+export default function NavigationDrawer() {
```
have this component take in a prop called `open` that controls the state of the drawer

collins-self replied:
Just to be clear, are you saying I just need to add that prop for future use or actually implement a solution like on MUI's docs https://mui.com/material-ui/react-drawer/ -> Responsive Drawer

zaviermiller replied:
@collins-self for now just add the prop, im going to update the layout story to wire it all together.

collins-self commented:
### Updated with the changes requested
*The most recent commit makes the changes that were pointed out*

On line 11 and 71, I'm having a 'Hydration Issue: Error' When attempting to make the collapsable settings a smaller font, what other way can I go about this?

zaviermiller commented on =:
```diff
@@ -0,0 +1,104 @@
+          width: drawerWidth,
+        }}
+      >
+        <Box>
```
this needs to have prop `sx={{ width: drawerWidth }}` so that the drawer is sized correct, its weird.

collins-self replied:
Are you saying the <Box> needs the drawerwidth?

zaviermiller replied:
yea, when i was playing with it i noticed the size of the drawer element correct, but the displayed drawer width was not. when i checked out the example in the docs it showed that the box inside the drawer had the width set.

zaviermiller commented on =:
```diff
@@ -0,0 +1,104 @@
+        }}
+      >
+        <Box>
+          <Box sx={{ pr: 3, mt: 5 }}>{logo}</Box>
```
make this prop the following (had to figure this out after some experimentation):
```
sx={{ display: 'flex', px: 2, pt: 2, justifyContent: 'center' }}
```
`px` puts padding on both the left and right side of the container.
`pt` puts the same amount of padding on the top of the container (these two used together will help space the image correct)
`display: 'flex'` makes this container a flex box so that we can use
`justifyContent: 'center'` to finish centering the image.

zaviermiller commented on =:
```diff
@@ -0,0 +1,104 @@
+      src="/../public/images/CCAHT-Logo.png"
+      alt="CCAHT-logo"
+      height="70"
+      width="239"
```
add a prop to the `Image` defined like:
```
style={{ objectFit: 'contain' }}
```

this will use the `object-fit` CSS property to ensure the image has the correct aspect ratio.

zaviermiller commented on components/ListItem/index.tsx:
```diff
@@ -0,0 +1,21 @@
+import ListItemText from '@mui/material/ListItemText'
```
rename this component and its folder to `NavigationDrawerListItem`

zaviermiller commented on =:
```diff
@@ -0,0 +1,104 @@
+import React from 'react'
```
just noticed this isnt the navigation drawer component, heed my comments but delete this file.

collins-self replied:
I'm not sure what you mean, I'm supposed to delete what file? the NavigationDrawer?

zaviermiller replied:
in the project, there is a file called `=` in the root of the project. delete this file

collins-self replied:
oh lol I don't know how that happened. will do

zaviermiller commented on components/NavigationDrawer/index.tsx:
```diff
@@ -0,0 +1,84 @@
+          <Box sx={{ pr: 3, mt: 5 }}>{logo}</Box>
+          <br />
+          {/* Drawer using ListItem functional component*/}
+          <ListItem icon={<AssignmentLateOutlinedIcon />} text="Dashboard" />
```
these should function to route between pages. to do this, add a prop on the `ListItem` component you defined called `route: string`, and then pass the route for each item in to it.

inside the `ListItem` component, change the `ListItemButton` to the following:
`<ListItemButton component={Link} href={route}> ... </ListItemButton>`

collins-self replied:
Will do, I thought this specific funtionality was for my other task on setting up the routing. Nonetheless, I need to do it.

zaviermiller replied:
disregard.

zaviermiller reviewed (changes requested):
moving in the right direction, good job!!

collins-self commented:
### Updated Name and Logo Styles
Still not able to get the Typography functionality working as @andlrutt requested for the Collapsable, and there are a few outstanding changes that I need some clarification on.
**Unresolved:** 
[The Typography Request](https://github.com/hack4impact-utk/community-coalition-against-human-trafficking/pull/3#discussion_r1027544322)
[Open State](https://github.com/hack4impact-utk/community-coalition-against-human-trafficking/pull/3#discussion_r1029380064)
[Drawer Width Request](https://github.com/hack4impact-utk/community-coalition-against-human-trafficking/pull/3#discussion_r1032673718)
[File Deletion?](https://github.com/hack4impact-utk/community-coalition-against-human-trafficking/pull/3#discussion_r1032675199)
[Will make this Routing Change in the Routing PR](https://github.com/hack4impact-utk/community-coalition-against-human-trafficking/pull/3#discussion_r1032678324)

andlrutt commented:
> ### Updated with the changes requested
> _The most recent commit makes the changes that were pointed out_
> 
> On line 11 and 71, I'm having a 'Hydration Issue: Error' When attempting to make the collapsable settings a smaller font, what other way can I go about this?

here's an example of changing the font of a `<ListItemText>` component. You'll probably need some way to differentiate the fact that they are collapsable... Maybe a boolean as another prop?
![image](https://user-images.githubusercontent.com/71574118/204150804-8585985c-dc4f-4230-a14b-be4160e76e70.png)

collins-self commented:
### All changes have been addressed 
The current commit has all changes addressed. Added a check for collapsable drawer items to give them the appropriate styling, Added the drawer prop 'open' that currently just returns the drawer or an empty div (if true or false respectively), Added the drawerWidth to a Box container in the Drawer, and deleted an unnecessary file.

collins-self commented:
### Made Final Changes
Fixed icons and prop formatting. please review kthxbye

zaviermiller reviewed (approved):
perfect!

zaviermiller commented on components/NavigationDrawerListItem/NavigationDrawer/index.tsx:
```diff
@@ -0,0 +1,116 @@
+      <Drawer
+        variant="permanent"
+        sx={{
+          width: drawerWidth,
```
control the appearance of the drawer here with `open={open}`

zaviermiller replied:
Add a prop here `onClose` that calls `setDrawerOpen` with false:

```jsx
<Drawer ... onClose={() => setDrawerOpen(false)> ... </Drawer>
```

this will let the drawer close when someone clicks away

andlrutt commented on components/NavigationDrawerListItem/index.tsx:
```diff
@@ -0,0 +1,33 @@
+}
+
+function NavigationDrawerListItem({
+  collapsable = false,
```
No need to supply a default value here

andlrutt commented on components/NavigationDrawerListItem/index.tsx:
```diff
@@ -0,0 +1,33 @@
+  text,
+}: ListProps) {
+  return collapsable ? (
+    <div>
```
No need for this div

andlrutt commented on components/NavigationDrawerListItem/index.tsx:
```diff
@@ -0,0 +1,33 @@
+        <ListItemIcon>{icon}</ListItemIcon>
+        <ListItemText
+          primary={text}
+          primaryTypographyProps={{ variant: 'body2' }}
```
Instead of a ternary expression to render one component or another, a shorter way to code this `<NavigationDrawerListItem>` component would be to just supply `primaryTypographyProps={{ variant: collapsable ? 'body2' : 'body1' }}` to this component. That way, there are no repeated lines of code.

andlrutt commented on components/NavigationDrawerListItem/NavigationDrawer/index.tsx:
```diff
@@ -0,0 +1,116 @@
+              <NavigationDrawerListItem text="Items" collapsable={true} />
+              <NavigationDrawerListItem
+                text="Item Attributes"
+                collapsable={true}
```
this and the above instances of `collapsable` do not need to be set to a value, you can simply just put the word `collapsable`. Since it's a boolean, a value of true will be passed to the child for that prop. If the word is not present (like with the other drawer list items), a value of false will be passed to the child for the `collapsable` prop

andlrutt reviewed (changes requested):
One more small change, the file/folder structure should have:
`components/NavigationDrawer/`. Inside of that folder, there should be `components/NavigationDrawer/index.tsx` and `components/NavigationDrawer/NavigationDrawerListItem/index.tsx`

I appreciate your persistence on this! We're really only making small changes, the overall code is GREAT!

RudraPatel2003 reviewed (approved):
Looks solid to me, great work!
